script.js

```diff
@@ -0,0 +1,60 @@
+
+     
+    let pic = document.getElementById('cnvs');
+    let ctx= pic.getContext('2d');
+
+    changeCanvas(pic.width, pic.height)
+
+
+    function firstPic(matrix, n, typeOfColor){
+        
+        for(let x=0; x<n; x++){
+            for(let y=0; y<n; y++){
+                if(typeOfColor==='#'){
+                    ctx.fillStyle =`#${matrix[x][y]}` ;
+                } else if (typeOfColor==='rgba'){
+                    ctx.fillStyle = `rgba(${matrix[x][y][0]}, ${matrix[x][y][1]}, ${matrix[x][y][2]}, ${matrix[x][y][3]})`
+                }
+                ctx.fillRect((1/n)*x*pic.width, (1/n)*y*pic.height, (1/n)*pic.width, (1/n)*pic.height);
+                }  
+            }
+        }
+
+    function showImg(){
+        
+        let img=new Image();
+        img.src='image.png';
+        ctx.fillRect(0, 0, pic.width, pic.height)
+        
+        img.onload=function(){
+            ctx.drawImage(img, 0, 0 ,pic.width, pic.height );
+        }
+        }
+
+    function changeCanvas(scale_w, scale_h){
+
+        pic.width=scale_w;
+        pic.height=scale_h;
+
+        ctx.fillStyle='grey';
+        ctx.fillRect(0, 0, pic.width, pic.width);
+
+    }
+
+document.getElementById('task_1').addEventListener('click', function(){firstPic(matrix1, 4, '#')})
+document.getElementById('task_2').addEventListener('click', function(){firstPic(matrix2, 32, 'rgba')})
+document.getElementById('task_3').addEventListener('click', showImg );
+document.getElementById('scale_change').addEventListener('click', function(){
+   changeCanvas(document.getElementById('scale_w').value, document.getElementById('scale_h').value)
+})
```

Comment:
each click on the 'scale_change' leads to a DOM call to receive 'scale_w' and 'scale_h' inputs. better call them one time and store as a constant:
`const WIDTH_SCALE = document.getElementById('scale_w')
const HEIGHT_SCALE = document.getElementById('scale_h')
document.getElementById('scale_change').addEventListener('click', function(){
   changeCanvas(WIDTH_SCALE.value, HEIGHT_SCALE.value)
})`